code/controllers/subsystem/minimap.dm

```diff
@@ -832,60 +844,68 @@ SUBSYSTEM_DEF(minimaps)
 			old_map = get_tacmap_data_png(faction)
 			current_svg = get_tacmap_data_svg(faction)
 
-		if ("updateCanvas")
-			// forces state change, this will export the svg.
+		if("updateCanvas")
 			toolbar_updated_selection = "export"
 			updated_canvas = TRUE
 			action_queue_change += 1
 
-		if ("clearCanvas")
+		if("clearCanvas")
 			toolbar_updated_selection = "clear"
 			updated_canvas = FALSE
 			action_queue_change += 1
 
-		if ("undoChange")
+		if("undoChange")
 			toolbar_updated_selection = "undo"
 			updated_canvas = FALSE
 			action_queue_change += 1
 
-		if ("selectColor")
+		if("selectColor")
 			var/newColor = params["color"]
 			if(newColor)
 				toolbar_color_selection = newColor
 				toolbar_updated_selection = newColor
 			action_queue_change += 1
 
-		if ("onDraw")
+		if("onDraw")
 			updated_canvas = FALSE
```

Comment:
do we want to return after these options?

Reply:
As in the same prevention checks? We could - I just didn't think it necessary because they don't actually *do* anything outside of the UI; and it seems moot to ensure the UI has good values if they're messing with the UI already.

`updated_canvas` resets the announcement button in the UI so you have to click it twice (because the canvas needs to lose focus to apply its last change)
`toolbar_updated_selection` getting set to a non-color enum tells the CanvasLayer an action to perform.
`action_queue_change` getting set to any value that is not the previous value is to prevent the above being performed multiple times.